nailgun/entities.py

```diff
@@ -1487,7 +1506,7 @@ def __init__(self, server_config=None, **kwargs):
                 OperatingSystem,
                 null=True,
             ),
-            'parent': entity_fields.OneToOneField(HostGroup, null=True),
+            'ancestry': entity_fields.StringField(null=True),
```

Comment:
Hashes? I don't understand how hashes are relevant here.

The API documentation for POST and PUT requests says this:

> hostgroup[parent_id] 
> Optional, Nil Allowed.
> Value: number

I then tried making some POST requests, like so:

```python
>>> from nailgun import client, entities
>>> from nailgun.config import ServerConfig
>>> cfg = ServerConfig.get()
>>> path = entities.HostGroup().path()
>>> response = client.post(
...     path,
...     {'hostgroup': {'name': 'hg2', 'parent_id': 9}},
...     **cfg.get_client_kwargs()
... )
>>> response.status_code
200
>>> for field_name in ('parent', 'parent_id', 'parent_ids', 'parents'):                                   
...     field_name in response.json()
... 
False
False
False
False
>>> response.json()['ancestry']
'9'
>>> response = client.post(
...     path,
...     {'hostgroup': {'name': 'hg3', 'parent_id': response.json()['id']}},  # 18
...     **cfg.get_client_kwargs()
... )
>>> response.raise_for_status()
>>> response.json()['ancestry']
'9/18'
>>> response = client.post(
...     path,
...     {'hostgroup': {'name': 'hg4', 'parent_id': response.json()['id']}},  # 19
...     **cfg.get_client_kwargs()
... )
>>> response.raise_for_status()
>>> response.json()['ancestry']
'9/18/19'
```

As you can see, each `HostGroup` entity has a foreign key relationship to some other `HostGroup` entity. In other words, the `HostGroup` database table has a self-referencing foreign key. This foreign key is named "parent". It just so happens that when GETting a `HostGroup`, the IDs of all parents are returned as a string containing decimal IDs joined by forward slashes. (This is utterly stupid IMO. Why not return a list of integer IDs? Or, even better, just the ID of the direct parent?)

What should be done? Keep the "parent" foreign key field:

```python
'parent': entity_fields.OneToOneField(HostGroup, null=True)
```

Then, add some extra logic in to `read` that parses 'ancestry' and uses that information to populate the `parent` field.